### PR TITLE
Set .envfile path based on build type

### PR DESCRIPTION
**Pull request by maxhora**
- .env file for prod builds is moved to `deployment/.env` and copied to the distribution packages;
- Customize path to .env file based on build type.

Depends on changes in https://github.com/status-im/react-native-config/pull/2

Note: If to run app with .env file from the repo's root, app doesn't start correctly in this case (because of some settings).

### CODE REVIEW

File: desktop_files/package.json [hunk 1]
```diff
@@ -55,7 +55,7 @@
     "react-native": "git+https://github.com/status-im/react-native-desktop.git",
     "react-native-background-timer": "2.0.0",
     "react-native-camera": "0.10.0",
-    "react-native-config": "git+https://github.com/status-im/react-native-config.git",
+    "react-native-config": "git+https://github.com/MaxRis/react-native-config.git",
```
Comment (maxhora) [inline desktop_files/package.json]: This change will be removed before merge

File: ci/desktop.groovy [hunk 2]
```diff
@@ -187,6 +187,8 @@ def bundleMacOS(type = 'nightly') {
     sh 'chmod +x Status.app/Contents/Resources/ubuntu-server'
     sh 'cp ../desktop/bin/StatusIm Status.app/Contents/MacOS/Status'
     sh 'cp ../desktop/reportApp/reportApp Status.app/Contents/MacOS'
+    sh "cp ../deployment/.env  Status.app/Contents/Resources"
+    sh 'ln -sf ../Resources/.env Status.app/Contents/MacOS/.env'
```
Comment (gnl) [inline ci/desktop.groovy]: Personally I'd prefer that there are no hidden files under `deployment` and `Resources`, so I'd go with:
```
sh 'cp ../deployment/env  Status.app/Contents/Resources'
sh 'ln -sf ../Resources/env Status.app/Contents/MacOS/.env'
```
...and of course everywhere else `.env` is referenced.

**Review by gnl — APPROVED** [review]

**Review by vitvly — APPROVED** [review]

**Comment by churik** [comment]
Tested:
- creating new account
- restoring existing account

File: ci/desktop.groovy (hunk 2 above)
Comment (maxhora) [reply ci/desktop.groovy]: @gnl , makes sense, thanks!

Comment (maxhora) [reply ci/desktop.groovy]: But why not just name the file "env" and set the appropriate file path to be loaded on app start.

**Review by vkjr — APPROVED** [review]

**Pull request merged**